platforms/nuttx/src/px4/common/hrt_ioctl.c

```diff
@@ -336,20 +351,32 @@ hrt_ioctl(unsigned int cmd, unsigned long arg)
 
 	case HRT_UNREGISTER: {
 			px4_sem_t *callback_sem = *(px4_sem_t **)arg;
+			sq_entry_t *queued;
+			void *deleted = NULL;
 			struct usr_hrt_call *e;
 			irqstate_t flags;
 
 			flags = spin_lock_irqsave_wo_note(&g_hrt_ioctl_lock);
 
-			while ((e = (void *)sq_remfirst(&callout_queue))) {
+			sq_for_every(&callout_queue, queued) {
+				kmm_free(deleted);
```

Comment:
Should this be:

if (deleted) {
  kmm_free(deleted);
  deleted = NULL;
}

Reply:
could be... freeing a NULLptr should be ok with every free function though, so it is redundant to add if

Reply:
I think kmm_free has some kind of debugassert i.e. not posix. Fine by me either way.